src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.Tests.cs

```diff
@@ -124,5 +124,31 @@ public class TarReader_File_Tests : TarReader_File_Tests_Base
         [InlineData(TarEntryFormat.Gnu, TestTarFormat.oldgnu)]
         public void Read_Archive_LongPath_Over255(TarEntryFormat format, TestTarFormat testFormat) =>
             Read_Archive_LongPath_Over255_Internal(format, testFormat);
+
+        [Fact]
+        public void Read_NodeTarArchives_Successfully()
+        {
+            string nodeTarPath = Path.Join(Directory.GetCurrentDirectory(), "tar", "node-tar");
+            foreach (string file in Directory.EnumerateFiles(nodeTarPath, "*.tar", SearchOption.AllDirectories))
+            {
+                using FileStream sourceStream = File.Open(file, FileMode.Open, FileAccess.Read, FileShare.Read);
+                using var reader = new TarReader(sourceStream);
+
+                TarEntry? entry = null;
+                while (true)
+                {
+                    Exception ex = Record.Exception(() => entry = reader.GetNextEntry());
```

Comment:
First time I see Record.Exception being used.

What advantage does it have over simply letting this throw?

Reply:
The intent of `Record.Exception()` followed by `Assert.Null()` is to be explicit about asserting that 'the method call is not throwing with this fix' (which were throwing without the fix). It also avoids seemingly dangling statements to be optimized out by the JIT.

Reply:
If you're consuming or asserting something about entry, Name, and Type the jit cannot remove them, and you're implicitly testing that they don't throw.

Eg you can assert that Name is never null, and that if type is Directory then length is zero.

Reply:
Eg you could do more or less what the test code did and then you won't need to record the exception explicitly
```c#
var ms = new MemoryStream();               
Assert.NotEmpty(entry.Name);                     
Assert.True(Enum.IsDefined(entry.EntryType));               
Assert.True(Enum.IsDefined(entry.Format));         
if (entry.EntryType == TarEntryType.Directory)
    continue;
var ds = entry.DataStream;
                       
if (ds != null && ds.Length > 0)
{                            
    ds.CopyTo(ms);
}
```